src/python/TaskWorker/Actions/PreJob.py

```diff
@@ -535,6 +535,17 @@ def execute(self, *args):
         sleep_time = int(self.dag_retry)*60
         if old_time:
             sleep_time = int(max(1, sleep_time - old_time))
+        # Double check if it is HC job as they want to have finer submittion
+        if self.task_ad.get("CRAB_HC", '') and \
+           self.task_ad.get("CRAB_HC_JOB_TIMEOUT", '0') and \
+           self.task_ad.get("QDate", ''):
+            timeout = self.task_ad.get("CRAB_HC_JOB_TIMEOUT", '0')
+            sleep_time = int(self.job_id * int(timeout))
+            # Also we want to double check with QDate as QDate is when the task
+            # was submitted.
+            diffFromNow = int(int(self.task_ad.get("QDate", 0) - (int(time.time()) + sleep_time)))
```

Comment:
if only one knew what sleep_time is ! other then the perverse logic, what is old_time expected to represent ?